sphinx/application.py

```diff
@@ -370,8 +372,6 @@ def build(self, force_all: bool = False, filenames: List[str] = None) -> None:
                 os.unlink(envfile)
             self.events.emit('build-finished', err)
```

Comment:
Do we want to emit twice in case there was an error?

Reply:
Maybe the status/warncount code could be moved outside the try statement instead, so that `build-finished` is emitted only once but errors are still logged afterwards.

Reply:
I think the following code will not raise an error. So I moved them outside of try-block now.
Thank you for your comment.